### ISSUE TITLE
BUG - Deployment test phase hangs indefinitely once all the tests are completed (using Cypress)

### ISSUE THREAD

**Issue by jeececab** [aws-amplify/amplify-console#347]
When I try to deploy, the tests run as usual, but the test step hangs indefinitely once the tests are over, therefore the deployment never happens. Had no similar problem before. I did a deployment last week with no problem whatsoever. I didn't change the build settings. 

![bb](https://user-images.githubusercontent.com/33330556/71590311-fab8f180-2af5-11ea-8316-5156324dcc0f.png)

Build settings:

![fddfd](https://user-images.githubusercontent.com/33330556/71586556-a1969100-2ae8-11ea-98cc-0ac9f95c0600.png)

**Comment by kahdojay** [comment]
@jeececab are you using 3.4.1? Please see https://github.com/aws-amplify/amplify-console/issues/115#issuecomment-564810589

**Comment by jeececab** [comment]
> @jeececab are you using 3.4.1? Please see [#115 (comment)](https://github.com/aws-amplify/amplify-console/issues/115#issuecomment-564810589)

Hey! Thanks for the follow up,
Nah, I was using 3.7.0. It was working well until last week. I also tried 3.8.1 and no luck

**Comment by kahdojay** [comment]
@jeececab can you try 3.4.1? Let us know if you still experience the issue using that version of Cypress.

**Comment by neilbts** [comment]
I have the same problem and confirmed it exists on 3.8.1 as well as 3.4.1.

**Comment by danieletieghi** [comment]
Same problem for me (v3.8.1). I would try with 3.4.1, but now I'm blocked by another problem in the build phase.

**Comment by germain-receeve** [comment]
Same issue here, downgrading to cypress 3.4.1 didn't help.
Anybody figured out a way around?

**Comment by patleeman** [comment]
I have the same issue.

The build hangs on test, or gets past test and shows a deploy failed. I've been trying many different variations with no luck.

ARN `arn:aws:amplify:us-east-1:641747502493:apps/dd79gp2sncpp1`

**Comment by patleeman** [comment]
For those having issues see https://github.com/aws-amplify/amplify-console/issues/352#issue-546445910

**Comment by ben-elsen** [comment]
I am facing the same issue!

**Comment by trietjack** [comment]
I temporarily fix it by running cypress tests in the build phase. But you can't download the videos if tests fail

**Comment by jeececab** [comment]
@kahdojay Hi! No 3.4.1 didn't resolve the issue.

**Comment by cslogan-red** [comment]
As per @patleeman (#352) and our latest comment on #115 , the recent change in mochawesome-merge syntax is causing this behavior, the exception is being eaten and not re-thrown properly, as described on #115, each fix is currently pending release and will be GA next week. In the meantime updating the syntax of the final postTest command as described will get your tests working again.

**Comment by swaminator** [comment]
@jeececab @neilbts  @trietjack @ben-elsen Please verify the fix. Closing the issue.

**Comment by neilbts** [comment]
@swaminator  I confirmed it is working as of this morning. Thank you!

**Comment by loera140** [comment]
I came across this as my project started to hang while starting the development server.  
![Screen Shot 2020-05-02 at 10 40 10 AM](https://user-images.githubusercontent.com/31187251/80871460-981d6100-8c61-11ea-9b38-bc9f7ee2ecd6.png). For me I was able to narrow it down to something to do with react-scripts. I found that ```"react-scripts": "^3.4.1"``` could not start my app within Amplify. However, by pinning the version to ```"react-scripts": "3.4.0"``` I was able to run all the tests successfully. Hopefully, this helps somebody.

Just for reference:
![Screen Shot 2020-05-02 at 10 47 00 AM](https://user-images.githubusercontent.com/31187251/80871584-47f2ce80-8c62-11ea-962e-9ba5cd8802f1.png)

Also, I ran into this same problem on a Jenkins server and fixed it there as well with this change
